src/tir/ir/stmt_functor.cc

```diff
@@ -640,10 +641,44 @@ Stmt IRTransform(Stmt ir_node, const runtime::PackedFunc& f_preorder,
   return transform(std::move(ir_node));
 }
 
+#define DEFINE_BIOP_EXPR_PROMOTABLE_MUTATE_(OP)                                           \
+  PrimExpr VisitExpr_(const OP##Node* op) override {                                      \
+    PrimExpr a = this->VisitExpr(op->a);                                                  \
+    PrimExpr b = this->VisitExpr(op->b);                                                  \
+    if (a.same_as(op->a) && b.same_as(op->b)) {                                           \
+      return GetRef<PrimExpr>(op);                                                        \
+    } else {                                                                              \
+      if (a.dtype().code() == b.dtype().code() && a.dtype().bits() != b.dtype().bits()) { \
```

Comment:
IR substituion will eventually fail if `a.dtype() != b.dtype()` which is required by binary operators at construction time (`OP(a, b)`). By promoting the operand types automatically when their base dtype (e.g., int or float) is the same can pass the test case (which is a valid model and failed previously due to `int32` in pool2d's topi conflicts `int64` in GPU attributes.